homeassistant/components/broadlink/entity.py

```diff
@@ -20,6 +20,8 @@ def __init__(self, device):
     async def async_added_to_hass(self):
         """Call when the entity is added to hass."""
         self.async_on_remove(self._coordinator.async_add_listener(self._recv_data))
+        if self._coordinator.data:
+            self._recv_data()
```

Comment:
This will cause a double state write since there's always a default state write at the end of entity addition. Instead call the method that updates the entity state from the coordinator directly, eg `_update_state`, and let the default state write call write that data to the state.

Reply:
Thanks. Changed it.